### PR TITLE
Authentication and Logging

### PR DESCRIPTION
**Pull request by terminalstderr**
* Document GCS backend.
* Simplify client.py.
* Update setup.py to use 'entry_points'.
* version bump
* Add a --version flag.

### CODE REVIEW

File: .groovylintrc.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+            "enabled": false
+        },
+        "DuplicateStringLiteral": {
+            "enabled": false
+        },
+        "NestedBlockDepth": {
+            "enabled": false
+        }
+    }
+}
```
Comment (morrowc) [inline .groovylintrc.json:14]: Unclear to me if required, but github is upset about: 
  "no newline at end of file"

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "cSpell.words": [
+        "bgpdata",
+        "routeviews"
+    ]
+}
```
Comment (morrowc) [inline .vscode/settings.json:6]: same as above with missing newline

File: routeviews_google_upload/README.md [hunk 3]
```diff
@@ -5,19 +5,31 @@
...
+
 The simplest invocation of this tool is to upload a file to a target gRPC server.
 
     routeviews-google-upload --dest https://grpc.routeviews.org --file /bgpdata/2021.03/UPDATES/update.20210331.2345.bz2
 
 > Run the command with the `--help` argument to see all the expected and available arguments.
 
+### Google Cloud Storage server, with Authentication
+
+If the targeted gRPC server is backed by a Google Cloud Storage (GCS) instance, it may require authentication.
```
Comment (morrowc) [inline routeviews_google_upload/README.md:27]: I think the actual problem here is that the cloudrun instance(s) require authentication (over https), before accepting the gRPC connection (which also arrives over https).

This is 'rpc over http(s)', and is.. super confusing ;( (to me anyway!)

Comment (terminalstderr) [reply routeviews_google_upload/README.md:27]: Aha, I'll update the wording to indicate that cloudrun backend requires Auth (and forget I ever mentioned GCS...). Thanks!

**Comment by morrowc** [comment]
seems great if it works :) hehe!

**Review by morrowc — APPROVED** [review]

**Comment by yuhao-he** [comment]
Not sure if it's covered anywhere, but in case it's not - can we strip off the first slash from the file paths? e.g. `/bgpdata/...` -> `bgpdata/...`

**Comment by terminalstderr** [comment]
> Not sure if it's covered anywhere, but in case it's not - can we strip off the first slash from the file paths? e.g. `/bgpdata/...` -> `bgpdata/...` So that we can keep new files consistent with the ones already uploaded.

@yuhao-he, Aha, yes, that is definitely doable!

Also, I should get me access to the 'routeviews-google data store' -- that way I can be sure our integration will provide the correct 'filename'.

**Pull request merged**